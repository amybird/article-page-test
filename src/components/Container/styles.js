import styled from 'styled-components';

export const Box = styled.div`
    width: 100%;
    max-width: 1080px;
    margin: 0 auto;
`;
