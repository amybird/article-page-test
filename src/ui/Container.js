import React from 'react';
import { Box } from 'rebass';
import { css } from 'styled-components/macro';

const Container = props => (
  <Box
    {...props}
    mx="auto"
    css={css`
      width: 100%;
      max-width: 1080px;
    `}
  />
);

export default Container;
