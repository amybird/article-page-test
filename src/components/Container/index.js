import React from 'react';
import PropTypes from 'prop-types';
import { Box } from './styles';

const Container = ({ children }) => (
  <Box>{children}</Box>
);

Container.defaultProps = {
  children: null
};

Container.propTypes = {
  children: PropTypes.node
};

export default Container;
