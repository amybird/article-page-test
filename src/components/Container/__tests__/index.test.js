import React from 'react';
import renderer from 'react-test-renderer';
import Container from '../index';

describe('Container', () => {
  it('should render a container', () => {
    const tree = renderer
      .create(<Container>Testing a Container wrapper</Container>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
