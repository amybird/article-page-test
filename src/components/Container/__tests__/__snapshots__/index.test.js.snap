// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render a container 1`] = `
.c0 {
  width: 100%;
  max-width: 1080px;
  margin: 0 auto;
}

<div
  className="c0"
>
  Testing a Container wrapper
</div>
`;
